### ISSUE TITLE
[Question] WARNING: tokenization mismatch when changing llama2 to llama2_chinese(Atom_7B_chat)

### ISSUE THREAD

**Issue by crazycth** [haotian-liu/LLaVA#708]
### Question

I tried to finetune llava1.5 with llama2_chinese as llm

here is my config:
```bash
#!/bin/bash

deepspeed llava/train/train.py \
    --lora_enable True --lora_r 128 --lora_alpha 256 --mm_projector_lr 2e-5 \
    --deepspeed ./scripts/zero3.json \
    --model_name_or_path /remote-home/ThCheng/weights/Atom_7B_chat \
    --version  llava_llama_2 \
    --data_path new_file_100.json \
    --image_folder /remote-home/ThCheng/dataset \
    --vision_tower openai/clip-vit-large-patch14-336 \
    --mm_projector_type linear \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio pad \
    --group_by_modality_length False \
    --bf16 False \
    --fp16 True \
    --bits 16 \
    --output_dir ./checkpoints/llava-v1.5-13b-lora \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 50000 \
    --save_total_limit 1 \
    --learning_rate 2e-4 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 False \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to wandb

```

But I encounter the tokenization mismatch problem:

```bash
WARNING: tokenization mismatch: 381 vs. 385. (ignored)
WARNING: tokenization mismatch: 462 vs. 466. (ignored)
WARNING: tokenization mismatch: 699 vs. 702. (ignored)
```

I think this is  the problem related to chinese tokenizer , but How should I solve this problem 🤔

**Comment by crazycth** [comment]
@haotian-liu Hi Liu , I'm confused here and I want to use Atom_7B_chat instead of raw llama2

Is this the problem related to tokenizer ? Could you give me some advises on how to solve this ?

Really Thank you for your time and guidance ~

**Comment by crazycth** [comment]
I solve this problem by set eos token in tokenizer, solve

**Comment by a2382625920** [comment]
Hi, can you elaborate on your solution, I also used the new model for SFT on LLAVA, and the eos token I set to True was automatically set back to default during fine tuning, and I got this error:
**WARNING: tokenization mismatch: 198 vs. 200. (ignored)**
But it's still fine-tuning, it doesn't stop, but the loss display is always 0.0

**Comment by STARRY2001** [comment]
> I solve this problem by set eos token in tokenizer, solve

I am a newbie. Can you tell me where in which file to add eos? Thank you very much!

**Comment by mylesgoose** [comment]
> > I solve this problem by set eos token in tokenizer, solve
> 
> I am a newbie. Can you tell me where in which file to add eos? Thank you very much!

https://huggingface.co/meta-llama/Meta-Llama-3-70B/discussions/6/files#d2h-846292     }
  },
  "bos_token": "<|begin_of_text|>",
  "add_bos_token": true,
  "add_eos_token": false,
  "chat_template": "{% set loop_messages = messages %}{% for message in loop_messages %}{% set content = '<|start_header_id|>' + message['role'] + '<|end_header_id|>\n\n'+ message['content'] | trim + '<|eot_id|>' %}{% if loop.index0 == 0 %}{% set content = bos_token + content %}{% endif %}{{ content }}{% endfor %}{{ '<|start_header_id|>assistant<|end_header_id|>\n\n' }}",
  "clean_up_tokenization_spaces": true,
  "eos_token": "<|end_of_text|>",
    "attention_mask"
  ],
  "model_max_length": 1000000000000000019884624838656,
  "tokenizer_class": "LlamaTokenizer" file is called [tokenizer_config.json]
